From 87f83a8b1289e2364a840a8d03063627972f53e4 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sun, 4 Sep 2022 08:26:26 -0700
Subject: [PATCH] odmsign.func: fix ODMDATA and overlay DTB handling for Xavier

tegraflash.py expects the caller to pass --bldtb with the base
bootloader DTB so it can incorporate the ODMDATA and any DTB
overlays into the signed bootloader DTB.

Signed-off-by: Matt Madison <matt@madison.systems>
---
 bootloader/odmsign.func | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bootloader/odmsign.func b/bootloader/odmsign.func
index 9d2082a..f81c616 100644
--- a/bootloader/odmsign.func
+++ b/bootloader/odmsign.func
@@ -577,7 +577,7 @@ odmsign_ext_sign ()
 			SIGNARGS+=" --cust_info ${custinfofilename} "
 		fi
 	else
-		SIGNARGS+="--bl ${flashername} ${BCT} ${bctfilename}";
+		SIGNARGS+="--bl ${flashername} --bldtb ${tbcdtbfilename} ${BCT} ${bctfilename}";
 	fi
 	if [ "${CHIPID}" = "0x19" ]; then
 		SIGNARGS+=",${bctfile1name} ";
